src/main/scala/FirstAssignment.scala

```diff
@@ -0,0 +1,41 @@
+package de.bapiakula.sparkscalacourse
+
+object FirstAssignment {
+
+  case class Transaction(items: String, quantity: Int, pricePerUnit: Double)
+
+  def totalCostAfterDiscount(totalCost: Double, discountThreshold: Double, discountRate: Double): Double = {
+    if (totalCost >= discountThreshold)
+      totalCost - (totalCost * discountRate)
+    else
+      totalCost
+  }
+
+  def totalCostCalculator(transactions: List[Transaction]): Double = {
+    transactions.map(t => t.quantity * t.pricePerUnit).sum
```

Comment:
Suggestion: We could add the `t.quantity * t.pricePerUnit` to the `Transaction` case class. As a case class is immutable, we can use a `val`.